lib/resolve-value.js

```diff
@@ -41,12 +39,19 @@ var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal
 
 	// Resolve any var(...) substitutons
 	var isResultantValueUndefined = false;
-	resultantValue = resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		// Loop through the list of declarations for that value and find the one that best matches
-		// By best match, we mean, the variable actually applies. Criteria:
-		//		- is under the same scope
-		//		- The latest defined `!important` if any
-		var matchingVarDeclMapItem;
+	
+	// var() = var( <custom-property-name> [, <any-value> ]? )
+	// matches `name[, fallback]`, captures "name" and "fallback"
+	// See: http://dev.w3.org/csswg/css-variables/#funcdef-var
+	var match;
+	while (match = balanced('var(', ')', resultantValue)) {
+		var matchingVarDeclMapItem = undefined;
+
+		// split comma to find variable name and fallback value
+		match.body = match.body.split(',');
```

Comment:
Let's make this more robust and not assign back to itself

```js
// Split at the comma to find variable name and fallback value
// There may be other commas in the values so this isn't necessary just 2 pieces
var variableFallbackSplitPieces = match.body.split(',');
```

Reply:
done, changed it for better reading